src/main/java/org/motechproject/mots/service/JasperReportsExportService.java

```diff
@@ -97,6 +100,40 @@ public void generateReport(UUID templateId, String format, HttpServletRequest re
     }
   }
 
+
+  /**
+   * Loads report template with given id and loads data from db to fill template
+   * and exports report to {@link HttpServletResponse} response.
+   *
+   * @param reportName name of the template that will be used to generate the report
+   * @return report data source
+   */
+  public DataSource generatePdfReport(String reportName) {
```

Comment:
I think instead of duplicating the code you should extract part of the existing export method to reuse the code.